### PR TITLE
Implementing mockserver&login w dagger(&tests)

### PR DESCRIPTION
**Pull request by Phaedra144**
Szia Aze!

Please approve :)

### CODE REVIEW

File: app/src/main/java/com/greenfox/peridot/peridot_coz_android/activity/MainActivity.java [hunk 1]
```diff
@@ -1,18 +1,78 @@
...
+    }
+
+//    public void loginExistingUser(User user) {
+//        if (!(user.getUsername().equals(getSharedPreferences("userInfo", Context.MODE_PRIVATE).getString("username",""))
+//                && user.getPassword().equals(getSharedPreferences("userInfo", Context.MODE_PRIVATE).getString("password","")))) {
+//            Toast.makeText(this,"Wrong username/password", Toast.LENGTH_SHORT).show();
+//            Intent loginIntent = new Intent(MainActivity.this, LoginActivity.class);
+//            startActivity(loginIntent);
+//        }
+//    }
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/peridot/peridot_coz_android/activity/MainActivity.java]: @Phaedra144 pls remove commented code

File: app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java [hunk 2]
```diff
@@ -1,22 +1,30 @@
...
 
-public class MockService implements UserInterface {
+@Module
+public class MockService implements UserInterface{
 
     private static final String TAG = "MockService";
-
     public MockService() {}
 
+    @Provides
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java]: @Phaedra144 move this to a specific `ApiProvider` class which does only providing this

File: app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java [hunk 3]
```diff
@@ -1,22 +1,30 @@
...
+    public MockService provideMockService(){
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java]: @Phaedra144 dont provide a MockService, provide the UserInterface instead

File: app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java [hunk 4]
```diff
@@ -1,22 +1,30 @@
...
+
+import dagger.Module;
+import dagger.Provides;
 import retrofit2.Call;
 import retrofit2.Callback;
 import retrofit2.Response;
 
-public class MockService implements UserInterface {
+@Module
+public class MockService implements UserInterface{
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java]: @Phaedra144 pls rename Userinterface to ApiService

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java (hunk 4 above)
Comment (radaladar) [reply app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java]: done

File: app/src/main/java/com/greenfox/peridot/peridot_coz_android/activity/MainActivity.java (hunk 1 above)
Comment (radaladar) [reply app/src/main/java/com/greenfox/peridot/peridot_coz_android/activity/MainActivity.java]: done

Comment (adamgyulavari) [reply app/src/main/java/com/greenfox/peridot/peridot_coz_android/activity/MainActivity.java]: cant see here :(

File: app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java (hunk 4 above)
Comment (adamgyulavari) [reply app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java]: I think you havent pushed :)

File: app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java (hunk 2 above)
Comment (radaladar) [reply app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java]: done

File: app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java (hunk 3 above)
Comment (radaladar) [reply app/src/main/java/com/greenfox/peridot/peridot_coz_android/model/api/MockService.java]: done

**Comment by radaladar** [comment]
requested changes done @adamgyulavari

**Review by adamgyulavari — APPROVED** [review]

**Comment by adamgyulavari** [comment]
@radaladar pls resolve conflicts

**Comment by radaladar** [comment]
@adamgyulavari conflicts resolved

**Pull request merged**